ios/GatewayManager.m

```diff
@@ -111,4 +151,9 @@ - (void)createBraintreeTokenWithPayment:(PKPayment *_Nonnull)payment
 #endif
 }
 
+- (UIViewController *)authenticationPresentingViewController
```

Comment:
This is interesting, how does it work?

Reply:
I know how it doesn't work! Crashes when opening the modal if this isn't here. https://developer.apple.com/documentation/uikit/uiviewcontroller/1621407-presentedviewcontroller 

Maybe it is called from within the stripe call ?

Reply:
Ok I kind of understand, you are saying up above `withAuthenticationContext self` which is telling Stripe that the class that contains all these functions:

https://stripe.dev/stripe-ios/docs/Protocols/STPAuthenticationContext.html#/c:objc(pl)STPAuthenticationContext(im)authenticationPresentingViewController

Is GatewayManager.m

I guess for 3D secure it needs this function

Reply:
Let's keep this in mind, there are other methods in there for Safari which we may need for other things. Wonder if Stripe Connect uses this